File: src/main/java/com/yoghurt1131/weatherapi/domain/City.java
package com.yoghurt1131.weatherapi.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.io.Serializable;
import java.util.ArrayList;

@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class City implements Serializable {

    private String name;
    @JsonProperty("weather")
    private ArrayList<Weather> weathers;
    @JsonProperty("main")
    private Temperature temperature;

    public String extractWeather() {
        if (weathers.size() == 0) {
            return null;
        }
        return weathers.get(0).getStatus();
    }

    public double extractKelvin() {
        if (weathers.size() == 0) {
            return 0.0;
        }
        return temperature.getTemp();
    }
}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/domain/City.kt
package dev.yoghurt1131.weatherapi.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import dev.yoghurt1131.weatherapi.domain.input.valueobject.Weather

data class City(
    val name: String,
    @JsonProperty("weather")
    val weathers: List<Weather>,
    @JsonProperty("main")
    val temperature: Temperature
) {

        fun extractWeather() = if (weathers.isNotEmpty()) weathers.first().status else ""

        fun extractKelvin() = temperature.temp

        @JsonIgnore
        fun getCurrentWeather(): CurrentWeather {
                var currentWeather: CurrentWeather =
                        CurrentWeather(name, extractWeather(), extractKelvin(), weathers.first().getWeatherIconUrl())
                return currentWeather
        }

        fun buildWeather() = CurrentWeather(
                        name, extractWeather(), extractKelvin(), weathers.get(0).getWeatherIconUrl()
                )
}
